### ISSUE TITLE
`zig init-exe` / `zig init-lib` should not overwrite existing `src/main.zig`

### ISSUE THREAD

**Issue by hdorio** [ziglang/zig#9820]
As of today `zig init-exe` will stop with the following error when an existing build file exists
> existing build.zig file would be overwritten

It would be good to do the same thing for all other generated files (`src/main.zig` currently) or at least to skip them instead of overwritting them.

I was trying to convert a library project to an executable project, the `build.zig` file was renamed and so `zig init-exe` erased the content of my `main.zig` file without warning.

I patched my `zig` copy with the following:

```
diff --git a/src/main.zig b/src/main.zig
index 8b5fbba90..f6e9e646e 100644
--- a/src/main.zig
+++ b/src/main.zig
@@ -2751,6 +2751,12 @@ pub fn cmdInit(
         error.FileNotFound => {},
         else => fatal("unable to test existence of build.zig: {s}\n", .{@errorName(err)}),
     }
+    if (fs.cwd().access("src" ++ s ++ "main.zig", .{})) |_| {
+        fatal("existing src" ++ s ++ "main.zig file would be overwritten", .{});
+    } else |err| switch (err) {
+        error.FileNotFound => {},
+        else => fatal("unable to test existence of src" ++ s ++ "main.zig: {s}\n", .{@errorName(err)}),
+    }
     var src_dir = try fs.cwd().makeOpenPath("src", .{});
     defer src_dir.close();
```

**Comment by ifreund** [comment]
This seems like it would be a nice improvement, would you mind submitting that patch as a PR?

**Comment by hdorio** [comment]
@ifreund I'll let you discuss about what is the best thing to do with @g-w1 https://github.com/ziglang/zig/pull/9821#discussion_r714346459

**Comment by ifreund** [comment]
Implemented in  #9821
